src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -13484,7 +13484,8 @@ bool Unit::UpdatePosition(float x, float y, float z, float orientation, bool tel
         return false;
     }
 
-    bool const turn = (GetOrientation() != orientation);
+    // Check if angular distance changed
+    bool const turn = M_PI - fabs(fabs(GetOrientation() - orientation) - M_PI) > 0.001f;
```

Comment:
If current orientation (GetOrientation) is 2pi-0.000001 and orientation variable equals to 0.000001 your formula will give 2pi-0.000002 and this will of course be greater than eps, but their real distance is 0.000002 (if you take the shorter path)

At lowest you would have to check 
G3D::fuzzyGt(Position::NormalizeOrientation(GetOrientation() - orientation), 0.0f) && G3D::fuzzyGt(Position::NormalizeOrientation(orientation - GetOrientation()), 0.0f);